README.md

```diff
@@ -42,8 +42,59 @@ If you are migrating a service to DBT PaaS, [GOV.UK PaaS to DBT PaaS Migration](
     poetry run pre-commit install
     ```
 
+### Platform Helper architecture
+
+`platform-helper` is split into the following layers:
+
+Commands (UI) -> Domains -> Providers
+
+Code written for `platform-helper` should adhere to the following architecture, shown below.
+
+![platform-helper architecture](doc/platform-helper-architecture.png)
+
+#### Commands
+
+This is the (essentially) UI level of `platform-helper`
+
+We try to follow a noun/domain verb/action pattern, `platform-helper thing action`.
+
+E.g. `codebase` has the following commands (all to do with the codebase Domain):
+
+- `build`
+- `deploy`
+- `list`
+- `prepare`
+
+Each command has an associated Domain. 
+
+There should be no business logic within the command as this is implelemented in the Domain layer.
+
+CLI arguments for a command are pulled in via [click](https://click.palletsprojects.com/en/stable/) and passed to the Domain code along with any dependencies.
+
+#### Domains
+
+Domains are where the business logic for a given command lives.
+
+Each Domain is a class.
+
+Any logged information (`click.secho`) from the Provider level should live within the Domain level.
+
+Any common/reusable elements should be implemented in a Provider.
+
+#### Providers
+
+Providers are groups of similar logic that are linked by the resource/tool/thing they use rather than the result of their what actions is.
```

Comment:
```suggestion
Providers are groups of similar logic that are linked by the resource/tool/thing they use rather than the result of what their actions is.
```